React Router and Page Layouts

jvsluis opened the pull request:
Added react router package and setup the basic logic to move between pages
Added files for the 3 pages home, create, and browse 
Setup some simple styling and structure to the app
No real design work done in this commit just structuring

Note:
Should use conditional rendering to handle multi stage forms. I didn't add that in this pull.

Closes #8

jvsluis committed:
add: react router and page layout

added react router package and setup the basic logic to move between pages
added files for the 3 pages home, create, and browse
setup some simple styling and structure to the app
no real design work done in this commit just structuring

DH3773 reviewed (commented):
Thumbs up from me, good to pull if the others agree.

jvsluis committed:
fix: made css more tailwind friendly

Used the tailwind container and margin class
Used tailwind media queries to follow "mobile first" principle

Jasper-Nelligan commented on app/src/routes/create.js:
```diff
@@ -0,0 +1,10 @@
+import { Link } from "react-router-dom";
+
+export default function Create() {
```
Could we rename this to "Upload", and then create a route to another page called "Configure". It better fits the mockups that way

DH3773 replied:
Should the upload and configure (same goes for the printer side) be 2 separate pages entirely? Is a react page like a full on new html page, where when you click on a new route you load the selected page? If that is the case then shouldn't it be that the upload and configure are elements of one page that you can click between? To allow seamless transition and less loading.

Jasper-Nelligan replied:
Yes I suppose we could do that. I believe you're right that a new route demands loading a new page. I was just thinking about organization - it's possible that we'll add more steps to the create jobs process, so the question is how much code are we willing to support in a single page?

jvsluis replied:
The idea I had here is that each form stage shouldn't be a new route since it wouldn't make sense for someone to navigate directly to the second stage (which would be possible on separate routes). Further, the two stages would need to share state so keeping it on the same route simplifies it. Instead of two routes, the one route would use conditional rendering to flip between the "Upload" and "Configure" sections of the form.

Jasper-Nelligan commented on app/src/routes/browse.js:
```diff
@@ -0,0 +1,10 @@
+import { Link } from "react-router-dom";
+
+export default function Browse() {
```
There should be one more page after this. I was thinking we call it Confirm? Or maybe ConfirmJob to be more specific

Jasper-Nelligan reviewed (commented):
Code looks good 👍 Just need to add two routes so it follows the mockups
